components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -52,6 +53,21 @@ pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<In
     Ok(Some(i64::from(day)))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn to_days(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    if t.is_none() {
+        return Ok(None);
+    }
+    let t = t.as_ref().unwrap();
+    if t.is_zero() {
+        return ctx
```

Comment:
What if `NO_ZERO_DATE` flag is unset.

Reply:
Hi, how can I know the value of  `NO_ZERO_DATE`? Or I just can't use ` if t.is_zero() {` ? 
Thanks!

Reply:
You can capture the extra payload of RPN function and `extra.ret_field_type.sql_mode.constains(...)`, you can reference `components/tidb_query/src/expr/ctx.rs`

Reply:
Here I think it's irrelevant to `NO_ZER_DATE`. In `MySQL`, it just reports `NULL` no matter what status of `NO_ZER_DATE` is.